source_control/gh_keys_facts.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2013, André Paramés <git@andreparames.com>
+# Based on the Git module by Leonardo Comelli <leonardo.comelli@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: gh_keys_facts
+version_added: "2.1" 
+short_description: GitHub ssh keys facts.
+description:
+  - GitHub ssh keys facts.
+
+options:
+  user:
+    description: 
+      - Github username.
+    required: true
+
+  password:
+    description: 
+      - Github password. If 2FA is enabled for your account, you should generate a new personal access token. Required for get_key, add_key and remove_key 
+    required: false
+    default: none
+
+  key_id:
+    description: 
+      - The key id provided by Github.
+    required: false
+    default: none    
+
+author: Leonardo Comelli (@leocomelli)
+'''
+
+EXAMPLES = '''
+# Example from Ansible Playbooks
+
+# Gathering facts about all keys of a non-authenticated user (limit informations)
+- gh_keys_facts: user=leocomelli
+
+# Gathering facts about all keys of an authenticated user
+- gh_keys_facts: user=leocomelli password=secret
+
+# Gathering facts about a specific key of an authenticated user
+- gh_keys_facts: user=leocomelli password=secret key_id=8767854
+'''
+
+RETURN = '''
+output:
+  description: the data returned by Github API
+  returned: success
+  type: dict
+  sample: 
+    {
+        "created_at": "2016-03-22T14:49:01Z", 
+        "id": 12345678, 
+        "key": "ssh-rsa AAA...", 
+        "read_only": false, 
+        "title": "devel", 
+        "url": "https://api.github.com/user/keys/12345678", 
+        "verified": true
+    }
+'''
+
+import base64
+try:
```

Comment:
this should not currently be done at module level, basic.py takes care of importing json